src/callbacks.rs

```diff
@@ -204,25 +269,45 @@ pub extern fn pointer_scroll(_view: WlcView, _time: u32,
 }
 
 pub extern fn pointer_motion(_view: WlcView, _time: u32, point: &Point) -> bool {
-    pointer::set_position(*point);
-    if let Ok(action) = try_lock_action() {
-        match *action {
-            None => return false,
-            _ => {}
+    let mut result = false;
+    let mut maybe_action = None;
+    {
+        if let Ok(action_lock) = try_lock_action() {
+            maybe_action = action_lock.clone();
         }
-    } else {
-        warn!("Could not lock action");
-        return false;
     }
-    if let Ok(mut tree) = try_lock_tree() {
-        match tree.try_drag_active(*point) {
-            Ok(_) => return true,
-            Err(TreeError::PerformingAction(_)) => {},
-            Err(TreeError::Movement(MovementError::NotFloating(_))) => {},
-            Err(err) => error!("Error: {:#?}", err)
+    match maybe_action {
+        None => result = false,
+        Some(action) => {
+            if action.edges.bits() != 0 {
+                if let Ok(mut tree) = try_lock_tree() {
+                    // TODO Change to id of _view
+                    // Need to implement a map of view to uuid first though...
```

Comment:
Haha woops, implemented the map but didn't update this piece of code...that's important!